### PR TITLE
Update internal Linux CI jobs

### PR DESCRIPTION
**Pull request by matt-kwong**
Updating internal Linux jobs to be closer to our Jenkins jobs. Also, we're happy with fuzzers only running on oss-fuzz, so we can remove fuzzers from our internal CI.

### CODE REVIEW

File: tools/internal_ci/linux/grpc_portability_build_only.sh [hunk 1]
```diff
@@ -37,4 +37,4 @@ git submodule update --init
 
 # download docker images from dockerhub
 export DOCKERHUB_ORGANIZATION=grpctesting
-tools/run_tests/run_tests_matrix.py -f portability linux --internal_ci --build_only
+tools/run_tests/run_tests_matrix.py -f portability linux --internal_ci --build_only --internal_ci
```
Comment (jtattermusch) [inline tools/internal_ci/linux/grpc_portability_build_only.sh]: --internal_ci is there twice.

File: tools/internal_ci/linux/grpc_sanity.sh [hunk 2]
```diff
@@ -37,4 +37,4 @@ git submodule update --init
 
 # download base docker image from dockerhub
 export DOCKERHUB_ORGANIZATION=grpctesting
-tools/run_tests/run_tests.py -l sanity -c opt -t -x sponge_log.xml --use_docker --report_suite_name sanity_linux_opt
+tools/run_tests/run_tests.py -l sanity -c opt -t -x sponge_log.xml --use_docker --report_suite_name sanity_linux_opt --internal_ci
```
Comment (jtattermusch) [inline tools/internal_ci/linux/grpc_sanity.sh]: run_tests.py doesn't know about --internal_ci (only run_tests_matrix.py does).

**Comment by matt-kwong** [comment]
Issues:
Interop- #10969

**Comment by matt-kwong** [comment]
Thanks for catching those extra flags

**Pull request merged**